src/Model/Concerns/HasArrayAccess.php

```diff
@@ -11,7 +11,7 @@
  *     use \atk4\data\ModelArrayAccessTrait;
  * }.
  */
-trait ModelArrayAccessTrait
+trait HasArrayAccess
```

Comment:
Are you inspired with this naming by something?

- I think it is very common to put abstract (AbstractXxx, XxxBase, ...), interface (Ixxx) and Trait (XxxTrait) in the name.
- "HasXxx" is wrong to me. I much prefer "WithArrayAccessTrait".

Reply:
I would agree on `WithArrayAccess` but keep it under the Model traits folder `Model\Concerns`.
This folder already has the `HasActions` trait in it from the UserAction refactor.
As `WithArrayAccess` is for optional usage we can keep it there for devs to use if they want.

Reply:
- `Deprecated` NS?
- `WithActions` too?
- `Trait` suffix?

Reply:
No need to introduce a special NS even if trait is deprecated. We use the standard NS
`HasXxx` prefix is generally accepted
No `Trait` suffix

Reply:
source for 2nd two?

Reply:
Last 2 inspired by Laravel way. Talking about what others do ...

Reply:
https://www.php-fig.org/bylaws/psr-naming-conventions/

Reply:
If we set to follow PSR is another question

Reply:
I use `_` normally only from properties that are more hacks for internal implementations, ie. properties that should not be accessed even from private.

https://symfony.com/doc/current/contributing/code/standards.html#naming-conventions

The `Trait` suffix seems to very consistent across PHP community.

`Has` / `With` - I am still not sure with this one. `ArrayAccessTrait` `ArrayAccessibleTrait` can be also candidates. More research should be done to be really consitent with whole Atk as well as other PHP projects. I tend more to Symfony than Lavarel which contributes a lot to these standards.